pkg/files/directory.go

```diff
@@ -306,6 +306,7 @@ func copyDirectory(original Directory, new Directory) error {
 // NewRootDirectory returns a new root directory
 func NewRootDirectory() Directory {
 	return &memoryDirectory{
-		name: paths.RootPath(),
+		name:     paths.RootPath(),
+		PermBits: os.FileMode(0755),
```

Comment:
This here may be something we have to debate over. @HeavyWombat the permission we set here will as it defines the permissions for the folders that are created without being backed by assets.

Eg: have an asset:
```yml
- golang
  - main.go
```

and you write it to `~/this/path/does/not/exist/yet`. Pina-Golada will with this change create those folders using 0755

Reply:
I think the best idea is to rely on the system's umask. I had no idea how Golang does it, so I had to read the `os` package documentation.

```go
package main

import "os"

func main() {
	os.Mkdir("foobar", 0777)
}
```

```
➜  ~ umask
022
➜  ~
➜  ~ go run main.go
➜  ~
➜  ~ ls -ldh foobar
drwxr-xr-x  2 mdiester  staff    64B May 21 22:14 foobar
➜  ~
```

As advertised in the doc, the `Mkdir` function works as expected with the umask. So setting `0777` (full rights) will create the desired `0755` as most systems have a default umask of `022`. In case we run on systems with another umask, we would honour the settings accordingly.

Long story short, I think it is fair to set `0777` in `Mkdir` and let the system take care of the rest.

Reply:
Alright, we will default to 0777 permission and let the file system take care of umasking those permissions